criu/fsnotify.c

```diff
@@ -602,6 +606,31 @@ static int restore_one_inotify(int inotify_fd, struct fsnotify_mark_info *info)
 
 err:
 	close_safe(&target);
+
+	/*
+	 * As we close target after setting up an inotify on it, we create
+	 * an unwanted inotify event, we need to remove these event.
+	 */
+	if (!ret && (mask & IN_CLOSE_NOWRITE)) {
+		char buff[EVENT_BUF_SIZE];
+		struct inotify_event *ie = (struct inotify_event *)buff;
+		int ielen;
+		int len;
+
+		len = read(inotify_fd, buff, EVENT_BUF_SIZE);
```

Comment:
could you add a comment that we don't restore queued events, so we can read all events here.

Reply:
done.